File: scripts/remove_bg.py
from PIL import Image
import numpy as np

input_path = r"C:\Users\pavan\OneDrive\Desktop\UXI_Works\bakefactory_uxi\public\logo.png"
output_path = r"C:\Users\pavan\OneDrive\Desktop\UXI_Works\bakefactory_uxi\public\logo.png"

img = Image.open(input_path).convert("RGBA")
data = np.array(img, dtype=np.float32)

r, g, b, a = data[:,:,0], data[:,:,1], data[:,:,2], data[:,:,3]

# BFS flood fill from all 4 corners to detect background region
from collections import deque

height, width = img.size[1], img.size[0]
visited = np.zeros((height, width), dtype=bool)
mask = np.zeros((height, width), dtype=bool)

THRESHOLD = 220  # pixels brighter than this in all channels = background candidate

def is_background(y, x):
    return (r[y,x] >= THRESHOLD and g[y,x] >= THRESHOLD and b[y,x] >= THRESHOLD)

queue = deque()

# Seed all edge pixels
for x in range(width):
    if is_background(0, x): queue.append((0, x))
    if is_background(height-1, x): queue.append((height-1, x))
for y in range(height):
    if is_background(y, 0): queue.append((y, 0))
    if is_background(y, width-1): queue.append((y, width-1))

while queue:
    y, x = queue.popleft()
    if y < 0 or y >= height or x < 0 or x >= width:
        continue
    if visited[y, x]:
        continue
    visited[y, x] = True
    if not is_background(y, x):
        continue
    mask[y, x] = True
    queue.append((y+1, x))
    queue.append((y-1, x))
    queue.append((y, x+1))
    queue.append((y, x-1))

# Apply mask — make background pixels fully transparent
data[:,:,3] = np.where(mask, 0, data[:,:,3])

result = Image.fromarray(data.astype(np.uint8), "RGBA")
result.save(output_path, "PNG")
print("Done! Background removed cleanly.")

File: scripts/remove-bg.js
const sharp = require('sharp');
const path = require('path');

const inputPath = path.join(__dirname, '../public/logo.png');
const outputPath = path.join(__dirname, '../public/logo.png');

async function removeBackground() {
  const image = sharp(inputPath);
  const { data, info } = await image
    .ensureAlpha()
    .raw()
    .toBuffer({ resolveWithObject: true });

  const { width, height, channels } = info;
  const pixels = Buffer.from(data);

  const THRESHOLD = 230; // how "white" counts as background

  function isWhiteLike(idx) {
    const r = pixels[idx];
    const g = pixels[idx + 1];
    const b = pixels[idx + 2];
    const a = pixels[idx + 3];
    return a > 10 && r >= THRESHOLD && g >= THRESHOLD && b >= THRESHOLD;
  }

  function setTransparent(idx) {
    pixels[idx + 3] = 0;
  }

  function pixelIndex(x, y) {
    return (y * width + x) * channels;
  }

  // Flood fill from all 4 edges using a queue (BFS)
  const visited = new Uint8Array(width * height);
  const queue = [];

  // Seed from all edge pixels
  for (let x = 0; x < width; x++) {
    queue.push([x, 0]);
    queue.push([x, height - 1]);
  }
  for (let y = 0; y < height; y++) {
    queue.push([0, y]);
    queue.push([width - 1, y]);
  }

  while (queue.length > 0) {
    const [x, y] = queue.shift();
    if (x < 0 || x >= width || y < 0 || y >= height) continue;

    const vIdx = y * width + x;
    if (visited[vIdx]) continue;
    visited[vIdx] = 1;

    const pIdx = pixelIndex(x, y);
    if (!isWhiteLike(pIdx)) continue;

    setTransparent(pIdx);

    queue.push([x + 1, y]);
    queue.push([x - 1, y]);
    queue.push([x, y + 1]);
    queue.push([x, y - 1]);
  }

  await sharp(pixels, {
    raw: { width, height, channels },
  })
    .png()
    .toFile(outputPath);

  console.log('✅ Background removed with flood-fill. Saved to:', outputPath);
}

removeBackground().catch(console.error);
